java-frontend/src/main/java/org/sonar/java/resolve/TypeAndReferenceSolver.java

```diff
@@ -946,4 +957,16 @@ private static void associateReference(IdentifierTree tree, JavaSymbol symbol) {
     }
   }
 
+  private static class LambdaBlockReturnVisitor extends BaseTreeVisitor {
+
+    final Set<Type> types = new HashSet<>();
+
+    @Override
+    public void visitReturnStatement(ReturnStatementTree tree) {
```

Comment:
Does this visit all return statements inside the tree? If so, then I think we should check if the enclosing symbol of the return statement is the currently processed lambda or not.

```
.map(line0 -> { 
  int i = (new Func<int, int>( ii -> { return 1; }))();
  return line0.split("\\s"); 
})
```